From b5ad8b70e26f646d1bf8faca6155fe1f7149f627 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 17:27:56 -0400
Subject: [infiniband] Allow build of hw/ and ulp/ subdirectories independently

Message-id: <c1888df9b64ceede1ea3f65ce26d461d8fe6f09f.1411578494.git.dledford@redhat.com>
Patchwork-id: 96664
O-Subject: [Patch RHEL7 03/26] IB: Allow build of hw/ and ulp/ subdirectories independently
Bugzilla: 1107621
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Yann Droneaud <ydroneaud@opteya.com>

It is not possible to build only the drivers/infiniband/hw/ (or ulp/)
subdirectory with command such as:

    $ make ARCH=x86_64 O=./obj-x86_64/ drivers/infiniband/hw/

This fails with following error messages:

    make[2]: Nothing to be done for `all'.
    make[2]: Nothing to be done for `relocs'.
      CHK     include/config/kernel.release
      Using /home/ydroneaud/src/linux as source for kernel
      GEN     /home/ydroneaud/src/linux/obj-x86_64/Makefile
      CHK     include/generated/uapi/linux/version.h
      CHK     include/generated/utsrelease.h
      CALL    /home/ydroneaud/src/linux/scripts/checksyscalls.sh
    /home/ydroneaud/src/linux/scripts/Makefile.build:44: /home/ydroneaud/src/linux/drivers/infiniband/hw/Makefile: No such file or directory
    make[2]: *** No rule to make target `/home/ydroneaud/src/linux/drivers/infiniband/hw/Makefile'.  Stop.
    make[1]: *** [drivers/infiniband/hw/] Error 2
    make: *** [sub-make] Error 2

This patch creates a Makefile in hw/ and ulp/ and moves each
corresponding parts of drivers/infiniband/Makefile in the new
Makefiles.

It should not break build except if some hw/ drivers or ulp/ were
allowed previously to be built while CONFIG_INFINIBAND is set to 'n',
but according to drivers/infiniband/Kconfig, it's not possible. So it
should be safe to apply.

Signed-off-by: Yann Droneaud <ydroneaud@opteya.com>
Reviewed-by: Bart Van Assche <bvanassche@acm.org>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 729ee4efcc29c040a1729c058f03fae3cebc6035)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---
 create mode 100644 drivers/infiniband/hw/Makefile
 create mode 100644 drivers/infiniband/ulp/Makefile

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/Makefile b/drivers/infiniband/Makefile
index bf508b5..dc21836 100644
--- a/drivers/infiniband/Makefile
+++ b/drivers/infiniband/Makefile
@@ -1,18 +1,3 @@
 obj-$(CONFIG_INFINIBAND)  += core/
-obj-$(CONFIG_INFINIBAND_MTHCA)  += hw/mthca/
-obj-$(CONFIG_INFINIBAND_IPATH)  += hw/ipath/
-obj-$(CONFIG_INFINIBAND_QIB)  += hw/qib/
-obj-$(CONFIG_INFINIBAND_EHCA)  += hw/ehca/
-obj-$(CONFIG_INFINIBAND_AMSO1100) += hw/amso1100/
-obj-$(CONFIG_INFINIBAND_CXGB3)  += hw/cxgb3/
-obj-$(CONFIG_INFINIBAND_CXGB4)  += hw/cxgb4/
-obj-$(CONFIG_MLX4_INFINIBAND)  += hw/mlx4/
-obj-$(CONFIG_MLX5_INFINIBAND)  += hw/mlx5/
-obj-$(CONFIG_INFINIBAND_NES)  += hw/nes/
-obj-$(CONFIG_INFINIBAND_OCRDMA)  += hw/ocrdma/
-obj-$(CONFIG_INFINIBAND_USNIC)  += hw/usnic/
-obj-$(CONFIG_INFINIBAND_IPOIB)  += ulp/ipoib/
-obj-$(CONFIG_INFINIBAND_SRP)  += ulp/srp/
-obj-$(CONFIG_INFINIBAND_SRPT)  += ulp/srpt/
-obj-$(CONFIG_INFINIBAND_ISER)  += ulp/iser/
-obj-$(CONFIG_INFINIBAND_ISERT)  += ulp/isert/
+obj-$(CONFIG_INFINIBAND)  += hw/
+obj-$(CONFIG_INFINIBAND)  += ulp/
diff --git a/drivers/infiniband/hw/Makefile b/drivers/infiniband/hw/Makefile
new file mode 100644
index 0000000..e900b03
--- /dev/null
+++ b/drivers/infiniband/hw/Makefile
@@ -0,0 +1,12 @@
+obj-$(CONFIG_INFINIBAND_MTHCA)  += mthca/
+obj-$(CONFIG_INFINIBAND_IPATH)  += ipath/
+obj-$(CONFIG_INFINIBAND_QIB)  += qib/
+obj-$(CONFIG_INFINIBAND_EHCA)  += ehca/
+obj-$(CONFIG_INFINIBAND_AMSO1100) += amso1100/
+obj-$(CONFIG_INFINIBAND_CXGB3)  += cxgb3/
+obj-$(CONFIG_INFINIBAND_CXGB4)  += cxgb4/
+obj-$(CONFIG_MLX4_INFINIBAND)  += mlx4/
+obj-$(CONFIG_MLX5_INFINIBAND)  += mlx5/
+obj-$(CONFIG_INFINIBAND_NES)  += nes/
+obj-$(CONFIG_INFINIBAND_OCRDMA)  += ocrdma/
+obj-$(CONFIG_INFINIBAND_USNIC)  += usnic/
diff --git a/drivers/infiniband/ulp/Makefile b/drivers/infiniband/ulp/Makefile
new file mode 100644
index 0000000..f3c7dcf
--- /dev/null
+++ b/drivers/infiniband/ulp/Makefile
@@ -0,0 +1,5 @@
+obj-$(CONFIG_INFINIBAND_IPOIB)  += ipoib/
+obj-$(CONFIG_INFINIBAND_SRP)  += srp/
+obj-$(CONFIG_INFINIBAND_SRPT)  += srpt/
+obj-$(CONFIG_INFINIBAND_ISER)  += iser/
+obj-$(CONFIG_INFINIBAND_ISERT)  += isert/
-- 
1.7.1